src/main/java/com/atharvadholakia/password_manager/service/CredentialService.java

```diff
@@ -20,15 +20,34 @@ public CredentialService(CredentialRepository credentialRepository) {
   public Credential createCredential(String serviceName, String username, String password) {
     Credential credential = new Credential(serviceName, username, password);
 
-    log.info("Calling repository from service");
+    log.info("Calling repository from service to write to DB.");
     credentialRepository.save(credential);
 
     log.debug("Exiting createCredential from service");
     return credential;
   }
 
+  public Credential updateCredential(
+      String id, String serviceName, String username, String password) {
+    Credential credential = getCredentialById(id);
+
+    if (credential.getUsername() != null) {
+      credential.setUsername(username);
+    }
+    if (credential.getServiceName() != null) {
+      credential.setServicename(serviceName);
+    }
+    if (credential.getPassword() != null) {
+      credential.setPassword(password);
+    }
+
+    log.debug("Exiting updateCredential from service");
+    credentialRepository.save(credential);
```

Comment:
Isn't there an`update` method in repository? Or `save` automatically handle creating a new record or updating a record?

Reply:
The JPA repository does not provide a direct update method. That is why I have used the save method which will overwrite the credential if it exists.